projects/lang-portal/typing-tutor/db_setup.rb

```
@@ -0,0 +1,159 @@
+      exit 1
+    ensure
+      db.close if db
+    end
+  end
+
+  private
+
+  # Enables foreign key constraints in SQLite
+  #
+  # @param db [SQLite3::Database] The database connection
+  def self.enable_foreign_keys(db)
```

RuboCop (1.69.1)

[warning] `private` (on line 61) does not make singleton methods private. Use `private_class_method` or `private` inside a `class << self` block instead.

(Lint/IneffectiveAccessModifier)